references/joins.mdx

```diff
@@ -507,6 +507,69 @@ Specifying a `primary_key` and the join `relationship` allows Lightdash to:
+3. **cte_unaffected**: Only needed when calculating metrics that could be affected by the join relationships. In some cases, this CTE isn't required if you're not calculating any metrics that would suffer from fanout.
+4. **final**: Join the metrics CTEs together to create the complete result set.
+
+<Note>
+  Lightdash creates a separate cte_keys and cte_metrics for each table that contains metrics with fanouts. This is why you'll see names like "cte_keys_orders" and "cte_metrics_orders" in the example below, indicating they're specific to the orders table.
+</Note>
+
+##### Examples
+
+```sql
+# Step 1: Create cte_keys that determine the final grain of your results i.e. whatever we will group by.
+# Exclude fields that we use in aggregations to calculate metrics e.g. `"orders".amount`
+WITH cte_keys_orders AS (
+    SELECT DISTINCT
+      "orders".status AS "orders_status", -- grouping dimension
+      "orders".order_id AS "pk_order_id"  -- primary key
+    FROM "postgres"."jaffle"."payments" AS "payments"
+    LEFT OUTER JOIN "postgres"."jaffle"."orders" AS "orders"
+      ON ("orders".order_id) = ("payments".order_id)
+),
+# Step 2: Calculate metrics that are affected by fanouts
+cte_metrics_orders AS (
+    SELECT
+      cte_keys_orders."orders_status",
+      SUM("orders".amount) AS "orders_total_order_amount" -- order metric (affected by fanout)
+    FROM cte_keys_orders
+    LEFT JOIN "postgres"."jaffle"."orders" AS "orders" ON cte_keys_orders."pk_order_id" = "orders".order_id -- join with primary keys
+    GROUP BY 1 -- Note orders_status are grouping dimensions
+),
+# Step 3: Calculate metrics that are not affected by fanouts
+cte_unaffected AS (
+    SELECT
+      "orders".status AS "orders_status",
+      COUNT(DISTINCT "payments".payment_id) AS "payments_unique_payment_count" -- payment metric (NOT affected by fanout)
+    FROM "postgres"."jaffle"."payments" AS "payments"
+    LEFT OUTER JOIN "postgres"."jaffle"."orders" AS "orders"
+      ON ("orders".order_id) = ("payments".order_id)
+    GROUP BY 1 -- Note orders_status are grouping dimensions
+)
+# Step 4: Join the metrics CTEs together to create the final result with properly calculated metrics
+SELECT
+  cte_unaffected.*,
+  cte_metrics_orders."orders_total_order_amount" AS "orders_total_order_amount"
+FROM cte_unaffected
+INNER JOIN cte_metrics_orders ON (
+  cte_unaffected."orders_status" = cte_metrics_orders."orders_status" OR ( cte_unaffected."orders_status" IS NULL AND cte_metrics_orders."orders_status" IS NULL )
+)
+ORDER BY "orders_total_order_amount" DESC
+LIMIT 500
+```
+
+**See examples in our demo:**
+- [Single one-to-many join example](https://demo.lightdash.cloud/...)
+- [Chained one-to-many joins example](https://demo.lightdash.cloud/...)
+- [Multiple one-to-many joins example](https://demo.lightdash.cloud/...)
```

Comment:
@jesshitchcock could you create charts in the demo website that I can link here? 🙏